src/main/scala/io/iohk/ethereum/vm/EvmConfig.scala

```diff
@@ -75,18 +75,18 @@ case class EvmConfig(
     */
   def calcMemCost(memSize: UInt256, offset: UInt256, dataSize: UInt256): BigInt = {
     /** See YP H.1 (222) */
-    def c(m: UInt256): BigInt = {
+    def c(m: BigInt): BigInt = {
       val a = wordsForBytes(m)
       G_memory * a + a * a / 512
     }
 
-    val memNeeded = if (dataSize.isZero) UInt256.Zero else offset + dataSize
+    val memNeeded = if (dataSize.isZero) BigInt(0) else offset.toBigInt + dataSize.toBigInt
```

Comment:
We're still mixing the types `UInt256` and `BigInt`, especially `memNeeded <= memSize`. Now, I can't see anything wrong with, but I also didn't see anything wrong all `UInt256` implementation until the bug manifested. Why don't you want to simply change the types in the signature?